deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.kt

```diff
@@ -197,32 +238,180 @@ class DeepLinkProcessor(symbolProcessorEnvironment: SymbolProcessorEnvironment?
         ) + (element.getAnnotation(DEEP_LINK_CLASS)?.value?.value?.toList() ?: emptyList())
     }
 
-    private fun verifyMethod(methodElement: XMethodElement) {
-        if (!methodElement.isStatic()) {
+    private fun verifyCass(classElement: XTypeElement): Boolean {
```

Comment:
It might be worth setting up a structure where these validations methods can throw, and a root try catch for a DLDException is caught, so we can avoid boolean return types used to early terminate a function